test/K6/src/api/platform/authorization/maskinporten.js

```diff
@@ -0,0 +1,122 @@
+  var endpoint = config.buildMaskinPorteSchemaUrls(offeredByPartyId, 'revokeoffered');
+  var params = header.buildHeaderWithRuntimeAndJson(altinnToken, 'personal');
+  var body = [];
+  body.push(generatePolicyMatch(to, resourceid, orgnoOrPartyId));
+  var bodystring = JSON.stringify(body);
+  bodystring = bodystring.substring(1, bodystring.length-1)
+  return http.post(endpoint, bodystring, params);
+}
+
+/**
+ * POST call to revoke maskinportenschemas that have been received by the current party
+ * @param {*} altinnToken personal token for DAGL of receiver
+ * @param {*} coveredByPartyId the receiving party's partyid
+ * @param {*} from the offering organization's party id or organization number
+ * @param {*} resourceid the id of the resource to delegate
+ * @param {*} orgnoOrPartyId 'orgno' to set id to urn:altinn:organizationnumber
+ */
+export function revokeReceivedMaskinportenSchema(altinnToken, coveredByPartyId, from, resourceid, orgnoOrPartyId) {
+  var endpoint = config.buildMaskinPorteSchemaUrls(coveredByPartyId, 'revokereceived');
+  var params = header.buildHeaderWithRuntimeAndJson(altinnToken, 'personal');
+  var body = [];
+  body.push(generatePolicyMatch(from, resourceid, orgnoOrPartyId, true));
+  var bodystring = JSON.stringify(body);
+  bodystring = bodystring.substring(1, bodystring.length-1);
+  return http.post(endpoint, bodystring, params);
+}
+
+/**
+ * GET call to get maskinportenschemas that have been received by the current party
+ * @param {*} altinnToken personal token for DAGL
+ * @param {*} toPartyId party id or organization number of whom that offers the rule
+ * @param {*} to the receiving organization's party id or organization number
+ * @param {*} resourceid the id of the resource to delegate
+ * @param {*} orgnoOrPartyId 'orgno' to set id to urn:altinn:organizationnumber
+ */
+export function postMaskinportenSchema(altinnToken, offeredByPartyId, to, resourceid, orgnoOrPartyId) {
+  var endpoint = config.buildMaskinPorteSchemaUrls(offeredByPartyId, 'maskinportenschema');
+  var params = header.buildHeaderWithRuntimeAndJson(altinnToken, 'personal');
+  var body = [];
+  body.push(generatePolicyMatch(to, resourceid, orgnoOrPartyId));
+  var bodystring = JSON.stringify(body);
+  bodystring = bodystring.substring(1, bodystring.length-1)
+  return http.post(endpoint, bodystring, params);
+}
+
+/**
+ * function to build a policy match with action, user and resource details
+ * @param {*} toPartyId party id or organization number of whom that receives the rule
+ * @param {*} altinnAction read,write,sign
+ * @param {*} orgnoOrPartyId 'orgno' to set id to urn:altinn:organizationnumber
+ * @param {*} useFrom 'true', 'false'. If true, use 'from' in json, else use 'to'.
+ * @returns json object of a completed policy match
+ */
+function generatePolicyMatch(toPartyId, resourceid, orgnoOrPartyId, useFrom) {
+  var toId = 'urn:altinn:partyid';
+  if (orgnoOrPartyId == 'orgno') {
```

Comment:
in stead of having a custom properties for toPartyId and orgnoOrPartyId (which again needs to be intepreted by code)
couldn't this just have the input values:
toAttributeId
toAttrubuteValue

Reply:
Aha! I'll do that.

Reply:
But this can be improved in iterations as well, as we add more tests

Reply:
I moved this to a new issue: https://github.com/Altinn/altinn-access-management/issues/368